File: output/gpt_4o/avatar/direct_translation/java/python/codeforces_459_B.py

import sys
from collections import Counter

def main():
    input = sys.stdin.read
    data = input().split()
    N = int(data[0])
    A = list(map(int, data[1:]))
    
    A.sort()
    min_val = A[0]
    max_val = A[-1]
    
    count = Counter(A)
    mins = count[min_val]
    maxs = count[max_val]
    
    if min_val == max_val:
        print(f"{max_val - min_val} {mins * (mins - 1) // 2}")
    else:
        print(f"{max_val - min_val} {mins * maxs}")

if __name__ == "__main__":
    main()

File: output/gpt_4o/avatar/direct_translation/python/java/codeforces_459_B.java

import java.util.*;

public class codeforces_459_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        List<Integer> l1 = new ArrayList<>();
        String[] input = scanner.nextLine().split(" ");
        for (String s : input) {
            l1.add(Integer.parseInt(s));
        }
        Collections.sort(l1);
        int a = l1.get(l1.size() - 1);
        int b = l1.get(0);
        if (a == b) {
            System.out.println((a - b) + " " + (n * (n - 1) / 2));
        } else {
            long ac = l1.stream().filter(x -> x == a).count();
            long bc = l1.stream().filter(x -> x == b).count();
            System.out.println((a - b) + " " + (ac * bc));
        }
    }
}

